engine/access/rpc/backend/backend.go

```diff
@@ -223,25 +222,72 @@ func convertStorageError(err error) error {
 func executionNodesForBlockID(
 	blockID flow.Identifier,
 	executionReceipts storage.ExecutionReceipts,
-	state protocol.State) (flow.IdentityList, error) {
+	state protocol.State,
+	log zerolog.Logger) (flow.IdentityList, error) {
 
 	// lookup the receipts storage with the block ID
-	receipts, err := executionReceipts.ByBlockIDAllExecutionReceipts(blockID)
+	allReceipts, err := executionReceipts.ByBlockIDAllExecutionReceipts(blockID)
 	if err != nil {
 		return nil, fmt.Errorf("failed to retreive execution receipts for block ID %v: %w", blockID, err)
 	}
 
-	// collect the execution node id in each of the receipts
+	// execution result ID to execution receipt map to keep track of receipts by their result id
+	var identicalReceipts = make(map[flow.Identifier][]*flow.ExecutionReceipt)
+
+	// maximum number of matching receipts found so far for any execution result id
+	maxMatchedReceiptCnt := 0
+	// execution result id key for the highest number of matching receipts in the identicalReceipts map
+	var maxMatchedReceiptResultID flow.Identifier
+
+	// find the largest list of receipts which have the same result ID
+	for _, receipt := range allReceipts {
+
+		resultID := receipt.ExecutionResult.ID()
+		identicalReceipts[resultID] = append(identicalReceipts[resultID], receipt)
+
+		currentMatchedReceiptCnt := len(identicalReceipts[resultID])
+		if currentMatchedReceiptCnt > maxMatchedReceiptCnt {
+			maxMatchedReceiptCnt = currentMatchedReceiptCnt
+			maxMatchedReceiptResultID = resultID
+		}
+	}
+
+	mismatchReceiptCnt := len(identicalReceipts)
+	// if there are more than one execution result for the same block ID, log as error
+	if mismatchReceiptCnt > 1 {
```

Comment:
logging error for both cases: if 2 mismatch ERs are observed or if more than 2 are seen